docs/source/llms/transformers/large-models.rst

```diff
@@ -0,0 +1,238 @@
+
+.. warning::
+
+    Ensure you specify the correct task argument, as an incompatible task will cause the model to **fail at the load time**. You can check the valid task type for your model on the HuggingFace Hub.
+
+
+.. _transformers-save-pretrained-guide:
+
+Storage-Efficient Model Logging
+-------------------------------
+
+Typically, when MLflow logs an ML model, it saves a copy of the model weight to the artifact store.
+However, this is not optimal when you use a pretrained model from HuggingFace Hub and have no intention of fine-tuning or otherwise manipulating the model or its weights before logging it. For this very common case, copying the (typically very large) model weights is redundant while developing prompts, testing inference parameters, and otherwise is little more than an unnecessary waste of storage space.
+
+To address this issue, MLflow 2.11.0 introduced a new argument ``save_pretrained`` in the :py:func:`mlflow.transformers.save_model()` and :py:func:`mlflow.transformers.log_model()` APIs. When with argument is set to ``False``, MLflow will forego saving the pretrained model weights, opting instead to store a reference to the underlying repository entry on the HuggingFace Hub; specifically, the repository name and the unique commit hash of the model weights are stored when your components or pipeline are logged. When loading back such a *refernce-only* model, MLflow will check the repository name and commit hash from the saved metadata, and either download the model weight from the HuggingFace Hub or use the locally cached model from your HuggingFace local cache directory.
+
+Here is the example of using ``save_pretrained`` argument for logging a model
+
+.. code-block:: python
+
+    import transformers
+
+    pipeline = transformers.pipeline(
+        task="text-generation",
+        model="meta-llama/Meta-Llama-3.1-70B",
+        torch_dtype="torch.float16",
+    )
+
+    with mlflow.start_run():
+        mlflow.transformers.log_model(
+            transformers_model=pipeline,
+            artifact_path="model",
+            # Set save_pretrained to False to save storage space
+            save_pretrained=False,
+        )
+
+In the above example, MLflow will not save a copy of the **Llama-3.1-70B** model's weights and will instead log the following metadata as a reference to the HuggingFace Hub model. This will save roughly 150GB of storage space and reduce the logging latency significantly as well for each run that you initiate during development.
+
+By navigating to the MLflow UI, you can see the model logged with the repository ID and commit hash:
+
+.. code-block:: bash
+
+    flavors:
+        ...
+        transformers:
+            source_model_name: meta-llama/Meta-Llama-3.1-70B-Instruct
+            source_model_revision: 33101ce6ccc08fa6249c10a543ebfcac65173393
+            ...
+
```

Comment:
Can we add a note / link for the API to load the the weights from this reference logging for the purposes of deployment?

Reply:
We have that in the "OSS Model Registry or Legacy Workspace Model Registry" section. Will add a quick sentence to navigate there